gitlab (#1751)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud/cloud-configuring-dbt-cloud/connecting-gitlab.md

## Overview

> Connecting your GitLab account to dbt Cloud unlocks exciting and compelling functionality in dbt Cloud. Once your GitLab account is connected, you can:

Reviewer: nit: but this comes off as a lil' cheesy to me
```suggestion
Once your GitLab account is connected to dbt Cloud, you can:
```

Reviewer 2: @dataders  and @schottj how does this sound?
```suggestion
Connecting your GitLab account to dbt Cloud provides convenience and another layer of security to dbt Cloud:
```

> - Log into dbt Cloud via OAuth through GitLab.

Reviewer: is this only for Enterprise tier?

Author: I need to delete this line actually. You can't "log into dbt Cloud via OAuth through GitLab.", we only support that via Github it seems

> - Log into dbt Cloud via OAuth through GitLab.

Reviewer 2: ```suggestion
- Log into dbt Cloud using OAuth through GitLab.
```

- Import new GitLab repos with a couple clicks during dbt Cloud project setup.
- Clone repos using HTTPS rather than SSH.

> - Trigger [Continuous integration](cloud-enabling-continuous-integration-with-github) builds when merge requests are opened in GitLab.

Reviewer: is there a reason this user permissions bullet got dropped?
```suggestion
- Carry GitLab user permissions through to dbt Cloud IDE's git actions.
- Trigger [Continuous integration](cloud-enabling-continuous-integration-with-github) builds when merge requests are opened in GitLab.
```

Reviewer 2: @dataders, I think the OAuth bullet point replaces this one.

Author: I need to specify "Carry GitLab user permissions through to dbt Cloud IDE's git actions." is only for the enterprise plan. We should delete the "log into dbt Cloud via OAuth through GitLab" altogether. That's incorrect.

> The steps to integrate GitLab in dbt Cloud depends on your plan. 

Reviewer 2: ```suggestion
The steps to integrate GitLab in dbt Cloud depend on your plan:
```

> If you are on:

Reviewer 2: I think we can remove if we add a colon to previous sentence.
```suggestion
```

- the Developer or Team plan, read these [instructions](/connecting-gitlab#for-dbt-cloud-developer-and-team-tiers).
- the Enterprise plan, jump ahead to these [instructions](/connecting-gitlab#for-the-dbt-cloud-enterprise-tier).

> ## For dbt Cloud Developer and Team tiers

Reviewer: @runleonarun just flagging a potential usecase for `<tier/>` blocks similar to `<version/>` blocks. not sure how often this comes up though

Reviewer 2: @dataders I have been thinking of how to handle this situation in the docs UI (a second drop down? a ), but I think you are spot on that we'll need to deal with this use case.

> To connect your GitLab account, navigate to **Your Profile** settings, and select **Linked Accounts** from the left navigation pane. There you'll see a **link** to connect your GitLab account.

Reviewer 2: ```suggestion
To connect your GitLab account: 
1. Navigate to Your Profile settings by clicking the gear icon in the top right. 
2. Select **Linked Accounts** in the left menu. 
3. Click **Link** to the right of your GitLab account.
```

2. [Add the GitLab app to dbt Cloud](/connecting-gitlab#adding-the-gitlab-oauth-application-to-dbt-cloud).

Once the admin completes those steps, dbt Cloud developers need to:
1. [Personally authenticate with GitLab](cloud-configuring-dbt-cloud/connecting-gitlab#personally-authenticating-with-gitlab) from dbt Cloud.

### Setting up a GitLab OAuth application
We recommend that before you set up a project in dbt Cloud, a GitLab account admin set up an OAuth application in GitLab for use in dbt Cloud.

For more detail, GitLab has a [guide for creating a Group Application](https://docs.gitlab.com/ee/integration/oauth_provider.html#group-owned-applications).

In GitLab, navigate to your group settings and select **Applications**. Here you'll see a form to create a new application.

<Lightbox src="/img/docs/dbt-cloud/connecting-gitlab/gitlab nav.gif" title="GitLab application navigation"/>

In GitLab, when creating your Group Application, input the following:

| Field | Value |
| ------ | ----- |
| **Name** | dbt Cloud |
| **Redirect URI** | https://cloud.getdbt.com/complete/gitlab |
| **Confidential** | ✔️ |
| **Scopes** | ✔️ api |

> Note, if you're a dbt Cloud single tenant customer, you'll need to replace **cloud.getdbt.com** with the hostname of

Reviewer 2: ```suggestion
dbt Cloud single tenant customers need to replace **cloud.getdbt.com** with the hostname of
```

In dbt Cloud, input the following values:

| Field | Value |
| ------ | ----- |
| **GitLab Instance** | https://gitlab.com |
| **Application ID** | *copy value from GitLab app* |
| **Secret** | *copy value from GitLab app* |

Note, if you have a special hosted version of GitLab, modify the **GitLab Instance** to use the hostname provided for your organization instead - for example https://gitlab.yourgreatcompany.com/.

Once the form is complete in dbt Cloud, click **Save**.

You will then be redirected to GitLab and prompted to sign into your account. GitLab will ask for your explicit authorization: 

<Lightbox src="/img/docs/dbt-cloud/connecting-gitlab/GitLab-Auth.png" title="GitLab Authorization Screen" />

Once you've accepted, you should be redirected back to dbt Cloud, and your integration is ready for developers on your team to [personally authenticate with](/connecting-gitlab#personally-authenticating-with-gitlab).

### Personally authenticating with GitLab
dbt Cloud developers on the Enterprise plan must each connect their GitLab profiles to dbt Cloud, as every developer's read / write access for the dbt repo is checked in the dbt Cloud IDE.

> To connect your personal GitLab account, dbt Cloud developers should navigate to their settings and scroll to **Linked Accounts** under **Your Profile**.

Reviewer 2: ```suggestion
To connect a personal GitLab account, dbt Cloud developers should navigate to Your Profile settings by clicking the gear icon in the top right, then select **Linked Accounts** in the left menu.
```

> If your GitLab account is not connected, you’ll see "No connected account". Select **Link** to begin the setup process. You’ll be redirected to GitLab, and asked to authorize dbt Cloud in a grant screen.

Reviewer 2: ```suggestion
If your GitLab account is not connected, you’ll see "No connected account." Select **Link** to begin the setup process. You’ll be redirected to GitLab, and asked to authorize dbt Cloud in a grant screen.
```

Author: https://owl.purdue.edu/owl/general_writing/punctuation/quotation_marks/more_quotation_mark_rules.html#:~:text=Place%20the%20punctuation%20outside%20the,%22You%20must%20work%20harder%22%3F

I believe how I have it is correct.
